docs/source/plugins.rst

```diff
@@ -76,6 +76,31 @@ View results at http://localhost:5000. You should see a newly-created run with a
     .. image:: ./_static/images/quickstart/quickstart_ui_screenshot.png
 
 
+Use Plugin for Client Side Authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+MLflow provides ``RequestAuthProvider`` plugin to customize auth header for outgoing http request.
+
+To use it, implement the ``RequestAuthProvider`` class and override the ``get_name`` and ``get_auth`` methods.
+``get_name`` should return the name of your auth provider, while ``get_auth`` should return the auth object
+that will be added to the http request.
+
+Once you have the implemented request auth provider class, register it in the ``entry_points`` and install the plugin.
```

Comment:
Can we add an example provider?

Reply:
As we're relying on this feature to be released before we released our plugin, at this moment we don't have a link for it. Is it good to add some dummy example just to show how to implement this abstract class?

Reply:
A dummy example would be helpful!

Reply:
Just updated